primitives/src/runtime_api.rs

```diff
@@ -0,0 +1,20 @@
+use crate::constants::transaction::MaxExtrinsicSize;
+use codec::Codec;
+use frame_support::BoundedVec;
+use sp_weights::Weight;
+
+sp_api::decl_runtime_apis! {
+	pub trait FeeEstimationApi<AccountId, AssetId, Balance>
+	where
+		AccountId: Codec,
+		AssetId: Codec,
+		Balance: Codec,
+	{
+		fn estimate_fee_payment(weight: Weight, account_id: AccountId) -> (AssetId, Balance);
```

Comment:
The result of both methods should return an Option.

In the actual implementation it seems it does unwrap_or in multiple places , so it can potentially return incorrect result, not the one you expect.

I would change the return value to be a struct. This would  help with potential future extension if we want  to include additional details.